retworkx/__init__.py

```diff
@@ -740,6 +740,78 @@ def _graph_random_layout(graph, center=None, seed=None):
     return graph_random_layout(graph, center=center, seed=seed)
 
 
+@functools.singledispatch
+def spring_layout(graph, pos=None, fixed=None, k=None, p=2,
+                  adaptive_cooling=True, niter=50, tol=1e-6,
+                  weight_fn=None, default_weight=1, scale=1,
+                  center=None, seed=None):
+    """
+    Position nodes using Fruchterman-Reingold force-directed algorithm.
+
+    The algorithm simulates a force-directed representation of the network
+    treating edges as springs holding nodes close, while treating nodes
+    as repelling objects, sometimes called an anti-gravity force.
+    Simulation continues until the positions are close to an equilibrium.
+
+    :param graph: Graph to be used. Can either be a
+        :class:`~retworkx.PyGraph` or :class:`~retworkx.PyDiGraph`.
+    :param dict (default=None) pos:
+        Initial node positions as a dictionary with node ids as keys and values
+        as a coordinate list. If None, then use random initial positions.
+    :param set (default=None) fixed: Nodes to keep fixed at initial position.
+        Error raised if fixed specified and pos not.
+    :param float (default=None) k:
+        Optimal distance between nodes. If None the distance is set to
+        1/sqrt(n) where n is the number of nodes.  Increase this value
+        to move nodes farther apart.
```

Comment:
```suggestion
    :param float  k:
        Optimal distance between nodes. If ``None`` the distance is set to
        :math:`\frac{1}{\sqrt{n}}` where :math:`n` is the number of nodes.  Increase this value
        to move nodes farther apart. (``default=None``)
```

Reply:
for some reason, this line can't pass the doc tests.

Reply:
Ah, that was my bad I forgot this docstring wasn't a raw string, so it's trying to escape `\f`, just replace `\frac` here with `\\frac` so it parses correctly (or alternatively you could start the docstring with `r"""` and leave it as `\frac`).

Reply:
oh thanks, that was subtle.